yt_dlp/extractor/tiktok.py

```diff
@@ -67,80 +92,87 @@ def _call_api_impl(self, ep, query, manifest_app_version, video_id, fatal=True,
-            'app_language': 'en',
-            'language': 'en',
+            'last_install_time': int(time.time()) - random.randint(86400, 1123200),
             'timezone_name': 'America/New_York',
+            'residence': 'US',
+            'app_language': 'en',
             'timezone_offset': '-14400',
-            'channel': 'googleplay',
-            'ac': 'wifi',
-            'mcc_mnc': '310260',
-            'is_my_cn': 0,
-            'aid': self._AID,
-            'ssmix': 'a',
-            'as': 'a1qwert123',
-            'cp': 'cbfhckdckkde1',
+            'host_abi': 'armeabi-v7a',
+            'locale': 'en',
+            'ac2': 'wifi5g',
+            'uoo': '1',
+            'op_region': 'US',
+            'build_number': self._APP_VERSION,
+            'region': 'US',
+            'ts': int(time.time()),
+            'iid': iid,
+            'device_id': random.randint(7250000000000000000, 7351147085025500000),
+            'openudid': ''.join(random.choices('0123456789abcdef', k=16)),
         }
 
     def _call_api(self, ep, query, video_id, fatal=True,
                   note='Downloading API JSON', errnote='Unable to download API page'):
-        if not self._WORKING_APP_VERSION:
-            app_version = self._configuration_arg('app_version', [''], ie_key=TikTokIE.ie_key())[0]
-            manifest_app_version = self._configuration_arg('manifest_app_version', [''], ie_key=TikTokIE.ie_key())[0]
-            if app_version and manifest_app_version:
-                self._WORKING_APP_VERSION = (app_version, manifest_app_version)
-                self.write_debug('Imported app version combo from extractor arguments')
-            elif app_version or manifest_app_version:
-                self.report_warning('Only one of the two required version params are passed as extractor arguments', only_once=True)
-
-        if self._WORKING_APP_VERSION:
-            app_version, manifest_app_version = self._WORKING_APP_VERSION
-            real_query = self._build_api_query(query, app_version, manifest_app_version)
-            return self._call_api_impl(ep, real_query, manifest_app_version, video_id, fatal, note, errnote)
-
-        for count, (app_version, manifest_app_version) in enumerate(self._APP_VERSIONS, start=1):
-            real_query = self._build_api_query(query, app_version, manifest_app_version)
+        if not self._WORKING_APP_IID:
+            if iid := self._configuration_arg('iid', [None], ie_key=TikTokIE)[0]:
+                self._WORKING_APP_IID = iid
+                self.write_debug(f'Imported app install id {iid!r} from extractor argument')
+
+        if self._WORKING_APP_IID:
+            real_query = self._build_api_query(query, self._WORKING_APP_IID)
+            return self._call_api_impl(ep, real_query, video_id, fatal, note, errnote)
+
+        max_tries = len(self._APP_INSTALL_IDS)
+        for count, iid in enumerate(random.sample(self._APP_INSTALL_IDS, max_tries), start=1):
```

Comment:
https://github.com/yt-dlp/yt-dlp/pull/9548/files#r1543188731 would simplify this to:

```suggestion
        iids = self._configuration_arg('iid', self._APP_INSTALL_IDS, ie_key=TikTokIE):
        max_tries = len(iids)
        for count, iid in enumerate(random.sample(iids, max_tries), start=1):
```